lib/ActivityStreams.js

```diff
@@ -103,22 +137,30 @@ ActivityStreams.prototype = {
     switch (msgName) {
```

Comment:
It looks like this switch body is doing the same thing for every response, you could just build a mapping between the event name "TOP_FRECENT_SITES_REQUEST" and its memoized data source this._memoized.getTopFrecentSites and then extract the rest out instead of repeating it.